### DISCUSSION TITLE
how can I call graph.getIDManager() in Python

### DISCUSSION THREAD

**Discussion by lesleyDeng** [JanusGraph/janusgraph#4662] [Q&A]
The **graph** object is imported **from gremlin_python.structure.graph import Graph** (according to https://docs.janusgraph.org/basics/connecting/python/), and it does not have a **getIDManager** method.

**Comment by lesleyDeng** [comment]
I need the IDManager to perform addV('label').property(T.id, graph.getIDManager().toVertexId(customized_id)) and to execute graph.getIDManager().fromVertexId(vertex_id) after querying. As the implementation of toVertexId() returns id << (partitionBits + USERVERTEX_PADDING_BITWIDTH),  I find an alternative solution that is to call addV('label').property(T.id, customized_id << 8) in Python. What potential impacts might there be?

**Reply by li-boxuan** [reply]
If you are using gremlin python library, you don't have getIDManager method. An alternative solution is to use https://docs.janusgraph.org/advanced-topics/custom-vertex-id/#custom-string-id

**Reply by li-boxuan** [reply]
> I find an alternative solution that is to call addV('label').property(T.id, customized_id << 8)

I think that should work too.
